Obsidian/Entities/Living.cs

```diff
@@ -15,6 +17,75 @@ public class Living : Entity, ILiving
     public int AbsorbtionAmount { get; set; }
 
     public Vector BedBlockPosition { get; set; }
+
+    public IReadOnlyDictionary<PotionEffect, PotionEffectData> ActivePotionEffects => activePotionEffects;
+
+    private readonly ConcurrentDictionary<PotionEffect, PotionEffectData> activePotionEffects;
+
+    public Living()
+    {
+        activePotionEffects = new ConcurrentDictionary<PotionEffect, PotionEffectData>();
+    }
+
+    public async override Task TickAsync()
+    {
+        foreach (var (potion, data) in activePotionEffects)
+        {
+            data.CurrentDuration--;
+            
+            if (data.CurrentDuration <= 0)
+            {
+                await RemovePotionEffectAsync(potion);
+            }
+        }
+    }
+
+    public bool HasPotionEffect(PotionEffect potion)
+    {
+        return activePotionEffects.ContainsKey(potion);
+    }
+
+    public async Task ClearPotionEffects()
+    {
+        foreach (var (potion, _) in activePotionEffects)
+        {
+            await RemovePotionEffectAsync(potion);
+        }
+    }
+    
+    public Task AddPotionEffectAsync(PotionEffect potion, int duration) => AddPotionEffectAsync(potion, duration, 0);
+
+    public Task AddPotionEffectAsync(PotionEffect potion, int duration, byte amplifier) =>
+        AddPotionEffectAsync(potion, duration, amplifier, true, true, false);
+
+    public async Task AddPotionEffectAsync(PotionEffect potion, int duration, byte amplifier, bool showParticles,
+        bool showIcon, bool isAmbient)
+    {
```

Comment:
This could've made use of [C#'s optional parameters](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/named-and-optional-arguments#optional-arguments) instead of creating method overloads.

Reply:
+1 on that one.